### PR TITLE
IMAGE-1184 debian hvm images could be built from single repo

### PR DESCRIPTION
**Pull request by mgerdts**
I've performed the tests described in tests/README.md on debian 8/9/10 on bhyve and kvm.  debian 10 is experiencing a long pause on first boot described in OS-8055.  That seems likely to be a PI issue.

### CODE REVIEW

**Comment by mgerdts** [comment]
This puts swap after /, which inhibits automatic partition growth by cloud-init.

File: debian-10/late-command [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+sed -i '2s/.*joyent.com.*//g' /etc/hosts
+
+# Tune Linux vm.dirty_background_bytes (IMAGE-439)
+# The following tuning causes dirty data to begin to be background flushed at
+# 100 Mbytes, so that it writes earlier and more often to avoid a large build
+# up and improving overall throughput.
+echo "Setting vm.dirty_background_bytes"
+echo "vm.dirty_background_bytes=100000000" >> /etc/sysctl.conf
+
+# Disable password auth. SSH logon is via ssh key only. A password is being set
```
Comment (danmcd) [inline debian-10/late-command]: Same here as in the centos one: s/logon/login/g

File: README.md [hunk 2]
```diff
@@ -1,42 +1,211 @@
...
-ISO image with the necessary packages and sdc-vmtools installed for deploying 
-on SmartOS and Joyent Public Cloud.
+<!--
+    Copyright 2020 Joyent, Inc.
+-->
 
-For more information on live-build see the [live-build manual](http://live-systems.org/manual/stable/html/live-manual.en.html)
+# Debian Image Builder
+
+This repo allows one to create custom Debian images for use with Triton.
```
Comment (danmcd) [inline README.md:13]: Should there be cross-referencing that Centos (and others?) should see the mi-centos-hvm repo?

File: README.md [hunk 3]
```diff
@@ -1,42 +1,211 @@
...
 
+This invokes
+[bits-upload.sh](https://github.com/joyent/eng/blob/master/tools/bits-upload.sh)
+to upload the image to manta and updates.joyent.com.
+
+
+## Default Settings For Images
+
+Each image has the following characteristics.  See
+`debian-<release>/ks.cfg` for details on which packages are included.
```
Comment (jasonbking) [inline README.md]: Is `ks.cfg` the correct file for Debian (the name suggests something kickstart related)?

File: README.md [hunk 4]
```diff
@@ -1,42 +1,211 @@
...
+* `cloud-init` is installed, as it is responsible for interacting with the
+  host's metadata service to configure networking, run user scripts, etc.  It
+  requires `pyserial`, but for "reasons" the cloud-init developers have avoided
+  adding pyserial as a dependency.
+* `cloud-init` requires configuration in
+  `/etc/cloud/cloud.cfg.d/90\_smartos.cfg` to only enable the SmartOS
+  datasource, among other things.
+* A `preseed/early\_command` (`debian-<release>/early-command`) is used to tail
+  the installation log and write it to `/dev/ttyS0`.  This script is run under
+  busybox, so avoid use of anything too fancy.  `qemu` runs in such a way that
```
Comment (jasonbking) [inline README.md]: Nit 'so avoid use' sounds a bit awkward (maybe "so avoid the use" ?)

File: debian-8/late-command [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+sed -i '2s/.*joyent.com.*//g' /etc/hosts
+
+# Tune Linux vm.dirty_background_bytes (IMAGE-439)
+# The following tuning causes dirty data to begin to be background flushed at
+# 100 Mbytes, so that it writes earlier and more often to avoid a large build
+# up and improving overall throughput.
+echo "Setting vm.dirty_background_bytes"
+echo "vm.dirty_background_bytes=100000000" >> /etc/sysctl.conf
+
+# Disable password auth. SSH logon is via ssh key only. A password is being set
```
Comment (danmcd) [inline debian-8/late-command]: s/logon/login/g

File: debian-9/late-command [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+sed -i '2s/.*joyent.com.*//g' /etc/hosts
+
+# Tune Linux vm.dirty_background_bytes (IMAGE-439)
+# The following tuning causes dirty data to begin to be background flushed at
+# 100 Mbytes, so that it writes earlier and more often to avoid a large build
+# up and improving overall throughput.
+echo "Setting vm.dirty_background_bytes"
+echo "vm.dirty_background_bytes=100000000" >> /etc/sysctl.conf
+
+# Disable password auth. SSH logon is via ssh key only. A password is being set
```
Comment (danmcd) [inline debian-9/late-command]: s/logon/login/g

File: imgmanifest.in [hunk 7]
```diff
@@ -0,0 +1,37 @@
+{
+  "v": "2",
+  "name": "@NAME@",
+  "version": "@DATE@",
+  "type": "zvol",
+  "cpu_type": "host",
+  "description": "@DESC@",
+  "homepage": "https://docs.joyent.com/images/linux/CentOS",
```
Comment (danmcd) [inline imgmanifest.in]: Copy-paste error:  Debian, right?

**Review by danmcd — COMMENTED** [review]
More nits.

File: debian-10/late-command (hunk 1 above)
Comment (mgerdts) [reply debian-10/late-command]: ok

File: README.md (hunk 2 above)
Comment (mgerdts) [reply README.md:13]: Added link to github search.

File: README.md (hunk 3 above)
Comment (mgerdts) [reply README.md]: s/ks/preseed/

File: README.md (hunk 4 above)
Comment (mgerdts) [reply README.md]: ok

File: debian-8/late-command (hunk 5 above)
Comment (mgerdts) [reply debian-8/late-command]: ok

File: debian-9/late-command (hunk 6 above)
Comment (mgerdts) [reply debian-9/late-command]: ok

File: imgmanifest.in (hunk 7 above)
Comment (mgerdts) [reply imgmanifest.in]: yep

**Review by danmcd — APPROVED** [review]
Thank you.

**Review by jasonbking — APPROVED** [review]

**Pull request merged**